lib/ret_web/channels/hub_channel.ex

```diff
@@ -871,6 +950,21 @@ defmodule RetWeb.HubChannel do
     end
   end
 
+  defp authorize(socket, :write_entity_state) do
+    with account <- Guardian.Phoenix.Socket.current_resource(socket),
+         :ok <- if(account === nil, do: :not_logged_in, else: :ok),
```

Comment:
I reached for the `with` statement here because I don't know how to write this function with nesting one `case` or `if` statement inside another, and another, and another. How would you write this function?